File: Python/high_scores.py
class HighScores():
    
    def __init__(self, scores):
        
        self.scores = scores
        
        pass
        
    def latest(self):
        last_score = len(self.scores)-1
        print (self.scores[last_score])
        latest = self.scores[last_score]
        return latest

    def personal_best(self):
        best_score = 0
        for i in self.scores:
            if best_score < i:
                best_score = i
        return best_score    
                
    
    
    def personal_top_three(self):
        top_scores = [0]
        for i in self.scores:
            if top_scores[0] < i:
                top_scores.insert(0, i)
            else:
                if top_scores[1] < i:
                    top_scores.insert(1, i)
                else:
                    if top_scores[2] < i:
                        top_scores.insert(2, i)
        if len(self.scores) < 3:
            del top_scores[len(self.scores):len(top_scores)]
        else:
            del top_scores[3:len(top_scores)]
        return top_scores  

File: Ruby/high_scores.rb
class HighScores
    def initialize(scores)
      @list_scores = []
      for score in scores do
        @list_scores.append(score)
      end
    end

    def scores
      return @list_scores
    end
  
    def latest
      return @list_scores[-1]
    end
  end

  def personal_best
    return @list_scores.max
  end

  def personal_top_three
    return @list_scores.max(3)
  end
end

scores = [30, 50, 20, 70]
score = HighScores.new(scores)
puts score.scores
puts score.latest
puts score.personal_best
puts score.personal_top_three

scores = [100, 0, 90, 30]
score = HighScores.new(scores)
puts score.scores
puts score.latest
puts score.personal_best
puts score.personal_top_three

scores = [10, 30, 90, 30, 100, 20, 10, 0, 30, 40, 40, 70, 70]
score = HighScores.new(scores)
puts score.scores
puts score.latest
puts score.personal_best
puts score.personal_top_three
